core/api/service/state/impl/state_api_impl.cpp

```diff
@@ -129,6 +129,17 @@ namespace kagome::api {
         [](common::BufferOrView &&r) { return r.intoBuffer(); });
   }
 
+  outcome::result<std::optional<uint64_t>> StateApiImpl::getStorageSize(
+      common::BufferView key,
+      const std::optional<primitives::BlockHash> &block_hash_opt) const {
+    auto at = block_hash_opt ? block_hash_opt.value()
+                             : block_tree_->getLastFinalized().hash;
+    OUTCOME_TRY(header, header_repo_->getBlockHeader(at));
+    OUTCOME_TRY(trie_reader, storage_->getEphemeralBatchAt(header.state_root));
+    OUTCOME_TRY(res, trie_reader->get(key));
```

Comment:
`tryGet` returns `optional`, return `nullopt` if there is no such key

Reply:
For some reason `tryGet`
`    OUTCOME_TRY(res, trie_reader->tryGet(key));
    return res ? std::make_optional(res->size()) : std::nullopt;`
 returns `nullopt` for the same key as 'get' returns valid `BufferOrView` containing some size.

Reply:
Fixed